src/check/constrain/generate/collection.rs

```diff
@@ -18,16 +19,45 @@ pub fn gen_coll(
     constr: &mut ConstrBuilder,
 ) -> Constrained {
     match &ast.node {
-        Node::Set { elements } | Node::List { elements } | Node::Tuple { elements } => {
+        Node::Set { elements } | Node::List { elements } => {
+            let res = gen_vec(elements, env, false, ctx, constr)?;
             constr_col(ast, env, constr, None)?;
-            gen_vec(elements, env, false, ctx, constr)
+            Ok(res)
         }
-
-        Node::SetBuilder { .. } => {
-            Err(vec![TypeErr::new(ast.pos, "Set builders currently not supported")])
+        Node::Tuple { elements } => {
+            let res = gen_vec(elements, env, env.is_def_mode, ctx, constr)?;
+            constr_col(ast, env, constr, None)?;
+            Ok(res)
         }
-        Node::ListBuilder { .. } => {
-            Err(vec![TypeErr::new(ast.pos, "List builders currently not supported")])
+
+        Node::SetBuilder { item, conditions } | Node::ListBuilder { item, conditions } => {
+            let conds_env = generate(item, &env.is_def_mode(true), ctx, constr)?;
+            if let Some(cond) = conditions.first() {
+                generate(cond, &conds_env.is_def_mode(false), ctx, constr)?;
```

Comment:
What if this is after the if below?
Ideally we want the error message there to be generated first before any potential ones within the condition. 

We must also add a sad test for if the first condition is not an in operation